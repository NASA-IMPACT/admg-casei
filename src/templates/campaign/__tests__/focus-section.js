import React from "react"
import { create } from "react-test-renderer"

import FocusSection from "../focus-section"

const testString = "test string"
const testArray = [
  {
    id: "a",
    shortname: "area a",
    longname: "area a",
  },
]

describe("Focus Section", () => {
  it("displays content", () => {
    const tree = create(
      <FocusSection
        id="focus"
        focus={testArray}
        geophysical={testArray}
        focusPhenomena={testString}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
