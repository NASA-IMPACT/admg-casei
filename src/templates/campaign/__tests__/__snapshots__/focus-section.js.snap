// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Focus Section displays content 1`] = `
<section
  className="inpage-nav"
  data-cy="focus-section"
  id="focus"
>
  <h2>
    Focus
  </h2>
  <div
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#303641",
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "display": "grid",
          "flex": "2.618",
          "gap": "1.5rem",
          "gridTemplateColumns": "minmax(0,1fr) minmax(0,1fr) minmax(0,1fr)",
          "padding": "2rem",
        }
      }
    >
      <div
        data-cy="focus-content"
      >
        <label
          data-cy="focus-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "x-small",
              "textTransform": "uppercase",
            }
          }
        >
          Focus Area
        </label>
        <div
          style={
            Object {
              "display": "grid",
              "gap": "1rem",
              "gridTemplateColumns": "repeat(auto-fill, minmax(min(120px, 100%), 1fr))",
              "paddingTop": "1rem",
            }
          }
        />
      </div>
      <div
        data-cy="focus-content"
      >
        <label
          data-cy="focus-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "x-small",
              "textTransform": "uppercase",
            }
          }
        >
          Geophysical Concept
        </label>
        <p
          data-cy="focus-content-text"
        >
          test string
        </p>
      </div>
      <div
        data-cy="focus-content"
      >
        <label
          data-cy="focus-content-label"
          style={
            Object {
              "borderBottom": "1px solid #9E9E9E",
              "color": "#9E9E9E",
              "display": "block",
              "fontSize": "x-small",
              "textTransform": "uppercase",
            }
          }
        >
          Focus Phenomena
        </label>
        <p
          data-cy="focus-content-text"
        >
          test string
        </p>
      </div>
    </div>
  </div>
</section>
`;
